### ISSUE TITLE
Bug: Notification History doesn't open app notifications as intended

### ISSUE THREAD

**Issue by ngwebb1** [crdroidandroid/issue_tracker#351]
### Issue type

Feature not working as intended

### Device

OnePlus 6 (enchilada)

### crDroid version

crDroid 10

### Exact version / Build date

10.2/2024-04-06

### Bug description

Tapping on a notification stored in Notification History does not launch its respective app. This effect seemingly applies to all apps except for system apps and apps downloaded from the F-Droid store.

### Steps to reproduce

1. Open Settings app.
2. Navigate to Notifications > Notification history.
3. Toggle on "Use notification history".
4. Await an app notification, tap on it to launch its respective app (or dismiss it), then view it Notification history.
5. Tap on it to relaunch. (This action fails and nothing happens when the notification is tapped.)

### Relevant log (logcat/build log)

```shell
04-07 20:51:58.443 E/SbnViewHolder(6119): Could not launch
04-07 20:51:58.443 E/SbnViewHolder(6119): android.app.PendingIntent$CanceledException
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at android.app.PendingIntent.send(PendingIntent.java:1076)
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at android.app.PendingIntent.send(PendingIntent.java:880)
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at com.android.settings.notification.history.NotificationSbnViewHolder.lambda$addOnClick$0(NotificationSbnViewHolder.java:113)
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at com.android.settings.notification.history.NotificationSbnViewHolder.$r8$lambda$Pcj3z3Ef1A1JG5Ro1BJsHSqiWA8(NotificationSbnViewHolder.java:0)
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at com.android.settings.notification.history.NotificationSbnViewHolder$$ExternalSyntheticLambda0.onClick(R8$$SyntheticClass:0)
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at android.view.View.performClick(View.java:7658)
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at android.view.View.performClickInternal(View.java:7635)
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at android.view.View$PerformClick.run(View.java:30189)
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at android.os.Handler.handleCallback(Handler.java:958)
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at android.os.Handler.dispatchMessage(Handler.java:99)
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at android.os.Looper.loopOnce(Looper.java:205)
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at android.os.Looper.loop(Looper.java:294)
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at android.app.ActivityThread.main(ActivityThread.java:8246)
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at java.lang.reflect.Method.invoke(Native Method)
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
04-07 20:51:58.443 E/SbnViewHolder(6119): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
04-07 20:51:58.898 E/TaskPersister(1616): File error accessing recents directory (directory doesn't exist?).
```

### Screenshots or videos

The logs (sourced from Matlog preinstalled on my build) provided had appeared when I tapped on the topmost Telegram notification in this screenshot.

![Screenshot_20240407-202934_Settings](https://github.com/crdroidandroid/issue_tracker/assets/81656695/29d63b3e-f3c1-4a37-9149-f6b8a5741546)

### Additional context

I have questioned users in the official crDroid Telegram group, yet no others have come across this issue. Even within my specific Telegram group for my device (maintained by Terminator_J), this is also true.

I appear to be the only one in my device's group running MinMicroG (sourced from [here](https://github.com/FriendlyNeighborhoodShane/MinMicroG-abuse-CI/releases/tag/2024.04.04), using [Terminator_J's crDroid installation instructions](https://gist.github.com/Terminator-J/172c8daa5a50312cbb1e39d81d482fec)) instead of GAPPS for my OnePlus 6, which could be the source of the problem. I have experimented with other ROMs for my device with the same MinMicroG package installed, and I don't have this problem.

Thanks for acknowledging this bug request. Please do inform me if I have done something wrong or if I need to clarify anything.

### Acknowledgements

- [x] I've checked device is officially supported (for device specific reports and not source related).
- [X] I'm running latest version available on crdroid.net for this device
- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by gwolf2u** [comment]
maybe needs crDroid update?
@Terminator-J

**Comment by ngwebb1** [comment]
> maybe needs crDroid update?
> @Terminator-J

I'm running on the latest official version from my device's maintainer...so I doubt it's the issue. Not sure what else I can try.

**Comment by gwolf2u** [comment]
latest available not equal to crdroid update I asked for on Jordan 
10.2 is old code

**Comment by ngwebb1** [comment]
What do you mean? I'm running 10.2 now. You're saying there's an update?

**Comment by gwolf2u** [comment]
> What do you mean? I'm running 10.2 now. You're saying there's an update?

above mention was not for you
wait for Jordans reply

**Comment by Terminator-J** [comment]
I don't think I've ever used that feature before (or gone to look for it).
Short, short version:
- The builds in TG group are using old ROM sources (synced before QPR2 merge went live with LineageOS), intentionally, because I'm trying to prepare an official "10.2 release 2" build for OTA to fix some ongoing issues in a sustainable way.
- The issue may have been fixed in crDroid 10.3 and 10.4 source, I just don't know yet because I haven't focused on fixing compilation errors in build tree with updated ROM sources.

If you can get a full logcat, not just the error paste URL, that would help. Apparently there's a directory missing according to that snippet you attached, and I'd imagine there's an sepolicy denial or access permission error earlier that might explain the problem.

Reboot, unlock, do something to give yourself a notification, try to access notification history so that it fails, and then capture logcat on computer right away. `adb logcat > logcat.log`, wait about 5-10 seconds, then ctrl+c to stop capturing, from a linux-style shell or powershell on windows should work. Send the generated logcat.log to me, so that we've also captured any boot-time denials.

I'll leave the issue open for now, in case it's still present after I start making 10.4 builds to test against.

**Comment by ngwebb1** [comment]
Thanks, J. Should I post the log cat here?

**Comment by Terminator-J** [comment]
You can PM me in telegram if you like.
Might be better in case of personal info in logs.

**Comment by gwolf2u** [comment]
this still relevant?

**Comment by ngwebb1** [comment]
This issue has been officially fixed in 10.3 according to the official dev team. I'm currently running a 10.4 build provided from my device's maintainer. This issue is to be closed momentarily.
